### PR TITLE
Chore: Add materialized view engine integration test

### PR DESCRIPTION
**Pull request by eakmanrq**

### CODE REVIEW

File: sqlmesh/core/engine_adapter/redshift.py [hunk 1]
```diff
@@ -100,5 +100,21 @@ def replace_query(
...
+    def _get_materialized_views(self, catalog_name: str, schema_name: str) -> exp.Expression:
+        return parse_one(
+            f"""            
+            SELECT
+                {catalog_name} AS catalog_name,
+                viewname AS name,
+                schemaname AS schema_name,
+                'MATERIALIZED_VIEW' AS type
+            FROM
+                pg_views
```
Comment (izeigerman) [inline sqlmesh/core/engine_adapter/redshift.py:112]: Provided that in `redshift` views are in the same table as materialized views, aren't they getting included here already https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/engine_adapter/base_postgres.py#L117-L123?

Comment (eakmanrq) [reply sqlmesh/core/engine_adapter/redshift.py:112]: Yes but just represented as a "view". We have this additional union in order to bring in views that are also materialized views.

Comment (eakmanrq) [reply sqlmesh/core/engine_adapter/redshift.py:112]: See this test demonstrating the current behavior: https://github.com/TobikoData/sqlmesh/blob/b0e7d2bd5c2d06ebb3975cb504582cbbe3e2d720/tests/core/engine_adapter/test_integration.py#L301-L306

Comment (izeigerman) [reply sqlmesh/core/engine_adapter/redshift.py:112]: wouldn't there be duplicates provided we're using `UNION ALL`?

Comment (eakmanrq) [reply sqlmesh/core/engine_adapter/redshift.py:136]: Talked and I now understand the issue and it is fixed. We also agreed to drop materialized view support for Redshift due to binding issues.

**Review by izeigerman — APPROVED** [review]

**Pull request merged**